src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2095,6 +2104,90 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
                             break;
                         }
                     }
+                case Instruction.AUTH:
```

Comment:
This is pretty stack heavy (x3 stackalloc, x3 byte[], several other locals), and fairly large. As stack costs are paid upfront at the method level and the Jit will bail on optimizing if a method gets too big, this will cause knock on effects to all tx whether or not they call AUTH.

Could it instead be moved out to a method `InstructionAuth`; see `InstructionLog`, `InstructionSLoad`, `InstructionSStore`, `InstructionRevert`, etc as examples

Reply:
Yes I wanted to refactor it to separate method. Do you think the zero padding and byte[] alloc should be done with stackalloc instead?

Reply:
Can look at that later, is more the method bloat of ExecuteCode that bothers me since its already vastly long which means the Jit is way passed trying to optimize local variables; so every one added even if just an `int` causes more space to be reserved and zeroed at the start of the method which adds to every call even if the code that uses that local is never run.

It will get even worse if going to add a `try`/`catch` for `_ecdsa.RecoverPublicKey`. Whereas if you move it to another method then the Jit will optimize that hard since its short and it won't add to the stack clearing and code size of the general dispatch method.

Reply:
`try/catch` will not help with `_ecdsa.RecoverPublicKey` since it will throw a CSE. 
Should i use `[SkipLocalsInit]` flag for the new method you think, or better to let JIT handle zeroing?